template.yml

```diff
@@ -32,9 +32,23 @@ Resources:
         CodeUri: src/handler/
         Handler: post.handler
         Runtime: nodejs18.x
+        Timeout: 20
         Events:
           PostLambdaApi:
             Type: Api
             Properties:
               Path: /v3/technical-records
               Method: post
+    PatchLambdaFunction:
+      Type: 'AWS::Serverless::Function'
+      Properties:
+        CodeUri: src/handler/
+        Handler: patch.handler
+        Runtime: nodejs18.x
+        Timeout: 20
+        Events:
+          PatchLambda:
+            Type: Api
+            Properties:
+              Path: /v3/technical-records/{systemNumber}/{createdTimestamp}
+              Method: patch
```

Comment:
In V1 it was a PUT however this wasn't exactly right as archiving a technical record shouldn't replace it and requiring the body to contain the entire tech record was unnecessary and not terribly secure as it allowed the client to misuse the endpoint to replace/add any field they liked just by changing the contents of the request. The endpoint now retrieves the technical record and only updates the fields that are necessary to archive the technical record thus making it a PATCH operation rather than a PUT operation.

Reply:
I believe PUT is going to be used for sending the entire technical record. There is a possibility of this endpoint being expanded to allow users to just send a single field that needs updating so to keep them distinct it was decided to use patch